server/src/main/scala/actors/ShiftsActorBase.scala

```diff
@@ -6,21 +6,22 @@ import akka.persistence._
 import akka.stream.scaladsl.SourceQueueWithComplete
 import com.trueaccord.scalapb.GeneratedMessage
 import drt.shared.CrunchApi.MillisSinceEpoch
+import drt.shared.FlightsApi.TerminalName
 import drt.shared._
 import org.slf4j.{Logger, LoggerFactory}
 import server.protobuf.messages.ShiftMessage.{ShiftMessage, ShiftStateSnapshotMessage, ShiftsMessage}
-import services.{OfferHandler, SDate}
 import services.graphstages.Crunch
+import services.{OfferHandler, SDate}
 
 import scala.concurrent.ExecutionContext.Implicits.global
-import scala.util.{Failure, Success, Try}
+import scala.util.Try
 
 
 case object GetState
 
 case object GetFeedStatuses
 
-case class GetPortState(from: MillisSinceEpoch, to: MillisSinceEpoch)
+case class GetPortState(from: MillisSinceEpoch, to: MillisSinceEpoch, maybeTerminalName: Option[TerminalName])
```

Comment:
this should possibly be a separate query, GetPortState and GetPortStateForTerminal

Reply:
Done